templates/theme.js

```diff
@@ -1,7 +1,18 @@
-// This is a global function also called from a script in ./rustdoc/body.html
-// which detects when the rustdoc theme is changed
 function applyTheme(theme) {
   document.documentElement.dataset.theme = theme;
 }
 
+window.addEventListener('storage', function (ev) {
+  if (ev.key === 'rustdoc-theme') {
+    applyTheme(ev.newValue);
+  }
+});
+
+// see ../static/storage-change-detection.html for details
+window.addEventListener('message', function (ev) {
+  if (ev.data && ev.data.storage && ev.data.storage.key === 'rustdoc-theme') {
+    applyTheme(ev.data.storage.value);
+  }
+});
```

Comment:
Do we need both of these listeners? Could we only listen to `storage` and get rid of the iframe?

Reply:
Then we miss out on knowing when this current window changes the theme, we would only update when another window changes it.